convert_sc_client_lookup.R

```diff
@@ -0,0 +1,271 @@
+      levels = c(0, 1), labels = c("No", "Yes")
+    ),
+    carer = factor(carer,
+      levels = c(0, 1), labels = c("No", "Yes")
+    ),
+    elderly_frail = factor(elderly_frail,
+      levels = c(0, 1), labels = c("No", "Yes")
+    ),
+    neurological_condition = factor(neurological_condition,
+      levels = c(0, 1), labels = c("No", "Yes")
+    ),
+    autism = factor(autism,
+      levels = c(0, 1), labels = c("No", "Yes")
+    ),
+    other_vulnerable_groups = factor(other_vulnerable_groups,
+      levels = c(0, 1), labels = c("No", "Yes")
+    ),
+    living_alone = factor(living_alone,
+      levels = c(0, 1, 9), labels = c("No", "Yes", "Not Known")
+    ),
+    support_from_unpaid_carer = factor(support_from_unpaid_carer,
+      levels = c(0, 1, 9), labels = c("No", "Yes", "Not Known")
+    ),
+    social_worker = factor(social_worker,
+      levels = c(0, 1, 9), labels = c("No", "Yes", "Not Known")
+    ),
+    type_of_housing = factor(type_of_housing,
+      levels = c(1:6),
+      labels = c(
+        "Mainstream", "Supported", "Long Stay Care Home",
+        "Hospital or other medical establishment", "Other",
+        "Not Known"
+      )
+    ),
+    meals = factor(meals,
+      levels = c(0, 1, 9), labels = c("No", "Yes", "Not Known")
+    ),
+    day_care = factor(day_care,
+      levels = c(0, 1, 9), labels = c("No", "Yes", "Not Known")
+    )
+  )
+
+
+
+# rename variables
+outfile <-
+  outfile %>%
+  rename(
+    sc_dementia = "dementia",
+    sc_mental_health_problems = "mental_health_problems",
+    sc_learning_disability = "learning_disability",
+    sc_physical_and_sensory_disability = "physical_and_sensory_disability",
+    sc_drugs = "drugs",
+    sc_alcohol = "alcohol",
+    sc_palliative_care = "palliative_care",
+    sc_carer = "carer",
+    sc_elderly_frail = "elderly_frail",
+    sc_neurological_condition = "neurological_condition",
+    sc_autism = "autism",
+    sc_other_vulnerable_groups = "other_vulnerable_groups",
+    sc_living_alone = "living_alone",
+    sc_support_from_unpaid_carer = "support_from_unpaid_carer",
+    sc_social_worker = "social_worker",
+    sc_type_of_housing = "type_of_housing",
+    sc_meals = "meals",
+    sc_day_care = "day_care"
+  )
+
+
+
+
+## save outfile ##
+
+outfile <-
+  outfile %>%
+  # reorder
+  select(
+    sending_location,
+    social_care_id,
+    sc_living_alone,
+    sc_support_from_unpaid_carer,
+    sc_social_worker,
+    sc_type_of_housing,
+    sc_meals,
+    sc_day_care
+  )
+
+
+
+# .zsav
+haven::write_sav(outfile,
+  paste0(
+    "/conf/sourcedev/Source_Linkage_File_Updates/", convert_year_to_fyyear(year),
+    "/Client_for_Source-20", convert_year_to_fyyear(year), ".zsav"
+  ),
+  compress = TRUE
+)
+
+# .rds file
+readr::write_rds(outfile,
+  paste0(
+    "/conf/sourcedev/Source_Linkage_File_Updates/", convert_year_to_fyyear(year),
+    "/Client_for_Source-20", convert_year_to_fyyear(year), ".rds"
+  ),
+  compress = "gz"
+)
```

Comment:
I made a function to grab the correct folder - [commit](https://github.com/Public-Health-Scotland/source-linkage-files/pull/152/commits/6a13590d3cfbd79a751d5ac0af6ddb102117ad4c) You could either cherry-pick that commit to your branch or wait till my branch is approved and merged then rebase/merge it in.